packages/plugin-page-view-tracking/src/page-view-tracking.ts

```diff
@@ -0,0 +1,75 @@
+import { IAmplitudeBrowser } from '@amplitude/analytics-browser';
+import {
+  BaseEvent,
+  BeforePlugin,
+  BrowserClient,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import { PluginCampaignTracker } from './plugin-campaign-tracker';
+import { PageTrackingBrowserOptions } from './typings/page-view-tracking';
+
+export class PageViewTrackingPlugin implements BeforePlugin {
+  name = 'page-view-tracking';
+  type = PluginType.BEFORE as const;
+
+  constructor(
+    private readonly instance: BrowserClient | IAmplitudeBrowser,
+    private readonly options: PageTrackingBrowserOptions = {},
+  ) {}
+
+  async setup(config: BrowserConfig): Promise<undefined> {
+    const attributionConfig = config.attribution;
+    const pageTrackingConfig = {
+      filter: attributionConfig?.trackPageViews ? 'onAttribution' : undefined,
+      ...this.options,
+    };
+
+    // Turn off sending page view event by "runAttributionStrategy" function
+    if (typeof attributionConfig === 'object') {
+      attributionConfig.trackPageViews = false;
+    }
+
+    const instance = this.instance;
+    // Share cookie storage with user session storage
+    const storage = config.cookieStorage as unknown as Storage<Campaign>;
```

Comment:
I think we should just use call `createFlexibleStorage` here instead of reusing a storage for a different purpose. It should yield the same storage provider

Reply:
Right. `createFlexibleStorage` is not exposed yet. Should we expose it?

Reply:
yes please. we can expose anything from analytics-client-common since it's an internal package and customers shouldnt be using it directly

Reply:
hmm sure but we do not repurpose storage instances in other places